test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,280 @@
+import { expect, jest } from '@jest/globals';
+import ThunderstoreMod from "../../../../src/model/ThunderstoreMod";
+import ThunderstoreVersion from "../../../../src/model/ThunderstoreVersion";
+import VersionNumber from "../../../../src/model/VersionNumber";
+import Game from "../../../../src/model/game/Game";
+import { getFullDependencyList } from "../../../../src/utils/DependencyUtils";
+import { GameInstanceType } from '../../../../src/model/game/GameInstanceType';
+import { GameSelectionDisplayMode } from '../../../../src/model/game/GameSelectionDisplayMode';
+import { PackageLoader } from '../../../../src/model/installing/PackageLoader';
+import ThunderstoreCombo from '../../../../src/model/ThunderstoreCombo';
+import ManifestV2 from '../../../../src/model/ManifestV2';
+
+// Mock IndexedDB
+jest.mock('dexie');
+
+function createCombo(params: {
+    name: string,
+    versionNumber: string,
+    dependencies?: string[],
+    categories?: string[]
+}): ThunderstoreCombo {
+    const combo = new ThunderstoreCombo();
+
+    const mod = new ThunderstoreMod();
+    mod.setName(`${params.name}`);
+    mod.setFullName(`author-${params.name}`);
+    mod.setDependencies(params.dependencies || []);
+    mod.setCategories(params.categories || []);
+    combo.setMod(mod);
+
+    const version = new ThunderstoreVersion();
+    version.setVersionNumber(new VersionNumber(params.versionNumber));
+    version.setName(`author-${params.name}-${params.versionNumber}`);
+    version.setDependencies(params.dependencies || []);
+    combo.setVersion(version);
+
+    return combo;
+}
+
+let mod: ThunderstoreCombo;
+let modWithDependency: ThunderstoreCombo;
+let modWithDependency2: ThunderstoreCombo;
+let modWithTheSameDependency: ThunderstoreCombo;
+let modWithTwoDependeciesThatHaveTheSameDependency: ThunderstoreCombo;
+let modWithDeepDependency: ThunderstoreCombo;
+let modPack: ThunderstoreCombo;
+let modWithLoopDependencyA: ThunderstoreCombo;
+let modWithLoopDependencyB: ThunderstoreCombo;
+
+let game: Game;
+
+const beforeSetup = () => {
+
+    game = new Game("", "RiskOfRain2", "", "", [], "", "", [], "",
+        GameSelectionDisplayMode.VISIBLE, GameInstanceType.GAME, PackageLoader.BEPINEX, []);
+
+    mod = createCombo({
+        name: "mod",
+        versionNumber: "1.0.0"
+    });
+
+    modWithDependency = createCombo({
+        name: "modWithDependency",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"]
+    });
+
+    modWithDependency2 = createCombo({
```

Comment:
Seems like a duplicate of `modWithTheSameDependency` and isn't  used?

Reply:
It is now `modWithAnotherDependency` (and has a different dependency), both are now used in dependencies in dexie. Not in tests though, but I think it's nice to keep the dexie mock's and test's mod list similar.
(IDK if there's some better way to do it than just repeating the same stuff)